tasks/unit_tests/version_tests.py

```diff
@@ -291,5 +292,6 @@ def test_on_release(self):
         self.assertEqual(get_matching_pattern(c, major_version="7", release=True), "7.55.0-rc.11")
 
     def test_on_branch(self):
-        c = MockContext()
+        c = MagicMock()
         self.assertEqual(get_matching_pattern(c, major_version="42", release=False), r"42\.*")
+        c.assert_not_called()
```

Comment:
I'm not sure to understand your modifications: Why is `MockContext` not suitable for this ? And is `assert_not_called` appropriate since the mock isn't called directly but `c.run()` is called ?

Reply:
Because we can't `assert_not_called` on them. In the context of this test (with `BUCKET_BRANCH` != nightly), `c.run()` should not be called so it's just an extra check